common/onion_message_parse.c

```diff
@@ -58,14 +58,27 @@ static bool decrypt_forwarding_onionmsg(const struct pubkey *blinding,
 	if (encmsg->path_id)
 		return false;
 
-	/* BOLT #4:
-	 * - SHOULD forward the message using `onion_message` to the next peer
-	 *   indicated by `next_node_id`.
+	/* BOLT-offers #4:
+	 * - if it is not the final node according to the onion encryption:
+	 *...
+	 *    - if `next_node_id` is present:
+	 *      - the *next peer* is the peer with that node id.
+	 *    - otherwise, if `short_channel_id` is present and corresponds to an announced short_channel_id or a local alias for a channel:
+	 *      - the *next peer* is the peer at the other end of that channel.
+	 *    - otherwise:
+	 *      - MUST ignore the message.
 	 */
-	if (!encmsg->next_node_id)
+	if (encmsg->next_node_id)
+		sciddir_or_pubkey_from_pubkey(next_node, encmsg->next_node_id);
+	else if (encmsg->short_channel_id) {
+		/* This is actually scid, not sciddir, but the type is convenient! */
+		struct short_channel_id_dir scidd;
+		scidd.scid = *encmsg->short_channel_id;
+		scidd.dir = 0;
+		sciddir_or_pubkey_from_scidd(next_node, &scidd);
+	} else
 		return false;
 
```

Comment:
This looks like useful information to log, no? like "ignoring message `blabla ..`" but this can be a followup PR